### PR TITLE
Make elixir-ts-mode-hook a customizable variable

### PR DESCRIPTION
**Pull request by sirikid**

### CODE REVIEW

**Comment by wkirschbaum** [comment]
@sirikid what problem are you trying to solve here? I have not seen this done on other modes and not sure what it achieves.

**Comment by sirikid** [comment]
> @sirikid what problem are you trying to solve here? I have not seen this done on other modes and not sure what it achieves.

Language mode hooks are intended for user customization thus they should be accessible through Custom.  I think most of the built-in hooks are customizable at this point.

**Comment by wkirschbaum** [comment]
@sirikid there is already a elixir-ts-mode hook which is part of deriving from prog-mode. what am I missing here? 

You can already do something like this: 
```
(add-hook 'elixir-ts-mode-hook (lambda () (subword-mode)))
```

**Comment by wkirschbaum** [comment]
I had a look at some other modes and saw lua-ts-mode is the only -ts-mode with this variable. Let me do some reason to see how it works. If you can give me an example it would also be great.

**Comment by sirikid** [comment]
> I had a look at some other modes and saw lua-ts-mode is the only -ts-mode with this variable. Let me do some reason to see how it works. If you can give me an example it would also be great.

[Custom][1] is an Emacs subsystem for interactive customization. [Here][2] you can see my `custom-file`, generated by this system, and then a screenshot of the interface.

On the left is the customization of the hook and on the right is the customization of another variable with a relatively complex structure (alist).

![image][3]

[1]: https://www.gnu.org/software/emacs/manual/html_node/emacs/Customization.html
[2]: https://git.sr.ht/~sokolov/dotfiles/tree/master/item/emacs/custom.el
[3]: https://github.com/wkirschbaum/elixir-ts-mode/assets/4840430/a574e048-f445-4f4e-8502-3816c2a0712d

**Comment by wkirschbaum** [comment]
I am guessing the proposed change allows you to use the customize menu instead of having to do it in your init.el file? Not many modes have this, so just trying to understand. The PR came with no motivation or description.

**Comment by sirikid** [comment]
> I am guessing the proposed change allows you to use the customize menu instead of having to do it in your init.el file? 

Yes.

> Not many modes have this, so just trying to understand.

I had a different experience.

> The PR came with no motivation or description.

Sorry, I thought it was a common knowledge.

**Comment by wkirschbaum** [comment]
> > Not many modes have this, so just trying to understand.
> 
> I had a different experience.

You can view the emacs source code and will find it is in only a handful of progmodes. 

> 
> > The PR came with no motivation or description.
> 
> Sorry, I thought it was a common knowledge.

Even so, its still good to give some motivation. 

I can't see a reason not to add it after having a look. Please add an appropriate comment as below then I can merge:
```elisp
(defcustom c-mode-hook nil
  "Hook called by `c-mode'."
  :type 'hook
  :group 'c)
 ```
 You don't need the comment either.

File: elixir-ts-mode.el [hunk 1]
```diff
@@ -613,6 +613,12 @@ Return nil if NODE is not a defun node or doesn't have a name."
     (save-excursion
       (newline 1 t))))
 
+(defcustom elixir-ts-mode-hook '()
+  "Hook run after entering Elixir mode."
```
Comment (wkirschbaum) [inline elixir-ts-mode.el:617]: "Hook run after entering `elixir-ts-mode'." its not Elixir mode.

File: elixir-ts-mode.el [hunk 2]
```diff
@@ -613,6 +613,12 @@ Return nil if NODE is not a defun node or doesn't have a name."
     (save-excursion
       (newline 1 t))))
 
+(defcustom elixir-ts-mode-hook '()
```
Comment (wkirschbaum) [inline elixir-ts-mode.el:616]: The convention is to specify 'nil' here.  Even though its the same it is still good to stick with conventions.

File: elixir-ts-mode.el [hunk 3]
```diff
@@ -613,6 +613,12 @@ Return nil if NODE is not a defun node or doesn't have a name."
...
+  "Hook run after entering Elixir mode."
+  :group 'elixir-ts
+  :type 'hook
+  :options (list #'eglot-ensure))
```
Comment (wkirschbaum) [inline elixir-ts-mode.el:620]: Please use the form '(eglot-ensure) as this seems to be the convention.

File: elixir-ts-mode.el (hunk 1 above)
Comment (sirikid) [reply elixir-ts-mode.el:617]: I used generated docstring, it uses the mode lighter

**Pull request merged**